### DISCUSSION TITLE
Supabase Refactor

### DISCUSSION THREAD

**Discussion by nfairbank** [mckaywrigley/chatbot-ui#1466] [Ideas]
I'm starting with this project as a base, but have some unique constraints for my deployed environment that preclude using Supabase out of the box (essentially I need to host totally sandboxed, with Kubernetes orchestration). To that end, I'm planning on a fairly major refactor that will take one of two forms: 

1) Simplifying the backend by replacing Supabase with base Postgres (and minimal other utilities), then refactoring next.js application code to use the simplified backend. 

2) Doing significant backend work by creating and orchestrating individual Supabase images with Kubernetes to replicate the needed  functionality, but hopefully leaving the next.js application code itself mostly intact. 

I'm leaning towards #2, but would love @mckaywrigley's input on what parts of Supabase the project actually uses (https://supabase.com/docs/guides/getting-started/architecture), to simplify the effort. 

I'm also happy to make this effort public if there others are interested in contributing to the process or benefiting from the outcome.

**Comment by jland-redhat** [comment]
I am also working with a group attempting to do an install of chatbot-ui and supabase on our Openshift environment based on [this chart](https://github.com/supabase-community/supabase-kubernetes) but we have found a lot of issues that need to be fixed at least for Openshift and we are struggle bussing through it pretty hard right now.

I honestly think option #1 is the best option, just have a basic DB set up and BYO the rest of your infrastructure. That is probably where we will end up at the end of the day.

But if we can get our version of the helm chart working we will put it in a public repo somewhere so others copy it.

**Comment by nfairbank** [comment]
@jland-redhat We've been pulling the thread on #2, for better or worse. Got to the point where we're building all images from scratch, and working on the helm orchestrations. If y'all have a helm chart together I would love to take a look; otherwise happy to share our progress.

**Comment by jland-redhat** [comment]
Okay, update time! We've been wrestling with this self-deployed Supabase setup. Their docs definitely leave something to be desired when you're not using Docker Swarm.

Despite that, we got a basic Supabase instance and the chat-ui working.  I tossed the Helm chart we used up here: https://github.com/jland-redhat/supabase-helm

Keep in mind it is still very much a WIP, but it should get the image up and running

**Reply by etherealmuse** [reply]
@jland-redhat  that's awesome, I was doing https://github.com/supabase-community/supabase-kubernetes got it working but ui is running into issues, going to try yours and see if it works for my case, I'm deploying on k8s cluster so lets see if it works well

**Comment by Jonneal3** [comment]
@nfairbank @stubclan @jland-redhat You guys should join the discord over here --  https://discord.gg/K5hkY333kz

**Comment by keithschacht** [comment]
If Supabase is a problem and you want more flexible deploy, you should consider this other chat bot:
https://github.com/allyourbot/hostedgpt

Very full featured, free, postgres is the only dependency, docker images and pre-built deploy scripts already built. Check the discussions for guidelines on Kubernetes, which has been figured out.

**Reply by spammenotinoz** [reply]
The interface is great, nice and simple, easy to use, suppose it is copied from open-webui, do you know if there any plans for multiple users like open-webui?

**Reply by keithschacht** [reply]
Yes, it already supports multiple users. It's all original implementation, but visually it was inspired by ChatGPT just like open-webui was. There's lots of similarity between all these projects but I thought I'd mention HostedGPT on this thread since it looks like there is a whole effort to simplify the database dependency for easier deploy and that's one way in which the HostedGPT project is really strong.
